packages/wallet/src/redux/protocols/indirect-funding/player-a/reducer.ts

```diff
@@ -275,15 +276,24 @@ function handleWaitForDirectFunding(
   return { protocolState, sharedData };
 }
 
-function createInitialSetupCommitment(allocation: string[], destination: string[]): Commitment {
+function createInitialSetupCommitment(
+  sharedData: SharedData,
+  allocation: string[],
+  destination: string[],
+): Commitment {
   const appAttributes = {
     proposedAllocation: [],
     proposedDestination: [],
     furtherVotesRequired: 0,
     updateType: UpdateType.Consensus,
   };
-  // TODO: We'll run into collisions if we reuse the same nonce
-  const nonce = 0;
+
+  const nonce = selectors.getLatestNonce(
```

Comment:
This has the effect of re-using the latest nonce, a nonce that has already been used.

Why not have a selector called `getNextNonce` to make this mistake more obvious?